Script to work with Jenkins in packaging and signing tacos

hab6 opened the pull request:
### Description
The script `package_sign_tacos.sh` is used with the instructions in the associated README as part of a Jenkins project that packages and signs Actian Tableau JDBC and ODBC connectors (tacos). The tacos are built from the latest code of the specified branch, using the `master` branch as the default.

### Jenkins Project
[Tableau_Connector](https://alm.actian.com/jenkins/ea/job/Tableau_Connector/)

### Location of Tacos
After the Jenkins build/script finishes, the JDBC and ODBC tacos are attached to the associated Jenkins build and can be downloaded for immediate use with Tableau tools.

### Taco Versions and Timestamps
The versions and timestamps of the JDBC and ODBC tacos can be found in the **Console Output** for the associated build.

hab6 committed:
Add script to package & sign tacos

hab6 committed:
README.md updates & formatting

hab6 committed:
Update README.md

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
```
Does this need to be bash or could it be a bourne shell?

hab6 replied:
Removed line that forced bash shell.

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+#!/bin/bash
+
```
Recommend adding some best practice, stop on error settings here (before any comments).

I don't have my full notes to hand (still on loaner laptop) but a good minimum is `set -e`. Linking to https://stackoverflow.com/a/2871034 which covers some others .

This could/should be done AFTER this get's merged. I.e. we know this works so lets get it in as-is, but then immediately follow up with sanity check additions and testing that.

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+fi
+
+# Set repo variables
+TABSDKREPO=https://github.com/tableau/connector-plugin-sdk.git
```
move to top of script, potentially make this a parameter (optional, with default)

hab6 replied:
Moved variable initialization to top of script (after comments).

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+
+# Set repo variables
+TABSDKREPO=https://github.com/tableau/connector-plugin-sdk.git
+ACTIANREPO=https://github.com/ActianCorp/actian_tableau_connector.git
```
move earlier in script

hab6 replied:
Moved variable initialization to top of script (after comments).

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+cd $BASEDIR
+
+# Create Python vitual environment
+python -m venv .venv
```
tempted to explicitly make this python3

clach04 replied:
and document here in a comment what version this has been used with.

hab6 replied:
Changed to explicitly use python3 with a comment of the python version tested.

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+cd $BASEDIR
+
+# Set environment variables needed for accessing certificate
+export SM_HOST=https://clientauth.one.digicert.com
```
move higher

clach04 commented on scripts/package_sign_tacos.sh:
```diff
@@ -0,0 +1,139 @@
+
+# Set environment variables needed for accessing certificate
+export SM_HOST=https://clientauth.one.digicert.com
+export SM_CLIENT_CERT_FILE=/usr/local/digicert/certs/Certificate_pkcs12.p12
```
also move higher, could be optional with default if omitted.

hab6 replied:
Moved variable initialization to top of script (after comments).

clach04 reviewed (commented):
Looks good, some recommendations for changes now and in the future.

Python3 doc note is the only thing that I think should change before being merged. The others could either happen now or later.

mianculovici reviewed (approved):
Approved.

hab6 committed:
Minor updates per comments in PR #48

hab6 commented:
Made these changes to the script `package_sign_tacos.sh` in the latest commit.
- Removed line that forced bash shell.
- Moved variable initialization to top of script (after comments).
- Change to explicitly use python3 with comment of version tested.

Updated the README, removing the `set +e`.

Future consideration
- Add "set -e" to script `package_sign_tacos.sh`.
- Have default values for variable initialization, allowing parameters to override.
